File: MatlabOctaveCodeSnippets/snip_signals_inpulse_train.m
Fs = 44100; %sampling frequency
Ts = 1/Fs; %sampling period
Timpulses = 0.25; %interval between impules in seconds
L=floor(Timpulses/Ts); %number of samples between impulses
N = 4; %number of impulses
impulseTrain=zeros(N*L,1); %allocate space with zeros
b=16; %number of bits per sample
amplitude = 2^(b-1)-1; %impulse amplitude, max signed int
impulseTrain(1:L:end)=amplitude; %generate impulses
writewav(impulseTrain,Fs,'impulses.wav','16r') %save WAVE RIFF

File: PythonCodeSnippets/snip_signals_inpulse_train.py
import numpy as np
from scipy.io.wavfile import write as wr

Fs = 44100
Ts = 1 / Fs
Timpulses = 0.25
L = np.floor(Timpulses / Ts)
N = 4
impulseTrain = np.zeros(int(N * L))
b = 16
amplitude = 2 ** (b - 1) - 1
N = 0
while N * L < len(impulseTrain):
    impulseTrain[N * int(L)] = amplitude
    N = N + 1
wr("Sinal.wav", Fs, impulseTrain)
